src/libraries/Microsoft.PowerFx.Core/Public/Config/OptionSet.cs

```diff
@@ -39,7 +39,7 @@ public OptionSet(string name, ImmutableDictionary<DName, DName> options)
         public OptionSet(string name, DisplayNameProvider displayNameProvider)
         {
             EntityName = new DName(name);
-            Options = displayNameProvider.LogicalToDisplayPairs;
+            Options = displayNameProvider.LogicalToDisplayPairs.ToImmutableDictionary();
```

Comment:
>LogicalToDisplayPairs.ToImmutableDictionary(); [](http://example.com/codeflow?start=42&length=46)

It would be nice to avoid this copy.  PVA is reporting perf issues here. 

I wonder if the Options property can just be an IEnumerable too? (technically a breaking change, but maybe ok?) 

Or we could make Options lazily computed.

Reply:
@lesaltzm Can you please advise on this?

Reply:
> > ```
> >     public ImmutableDictionary<DName, DName> Options { get; }
> > ```
> 
> @lesaltzm - could we break and make this IEnumerable?
> 
> Refers to: src/libraries/Microsoft.PowerFx.Core/Public/Config/OptionSet.cs:58 in [63ca8d3](https://github.com/microsoft/Power-Fx/commit/63ca8d393133881086674fdfea7ea45f72b0177b). [](commit_id = [63ca8d3](https://github.com/microsoft/Power-Fx/commit/63ca8d393133881086674fdfea7ea45f72b0177b), deletion_comment = False)

Yes, that should be a safe break, I think.

Reply:
Thanks @lesaltzm. Updated the code for Options.